src/deploy/functions/params.ts

```diff
@@ -66,29 +54,22 @@ export function resolveString(
   from: string | build.Expression<string>,
   paramValues: Record<string, ParamValue>
 ): string {
-  if (!isCEL(from)) {
-    return from;
-  }
   let output = from;
-  const paramCapture = /{{ params\.(\w+) }}/g;
+  const celCapture = /{{ (.+?) }}/;
+  let infiniteLoopGuard = 0;
   let match: RegExpMatchArray | null;
-  while ((match = paramCapture.exec(from)) != null) {
-    const referencedParamValue = paramValues[match[1]];
-    if (!referencedParamValue.legalString) {
+  while ((match = celCapture.exec(output)) != null) {
+    infiniteLoopGuard += 1;
+    if (infiniteLoopGuard > 100) {
       throw new FirebaseError(
-        "Referenced string parameter '" +
-          match[1] +
-          "' resolved to non-string value " +
-          referencedParamValue
+        `CEL string expression ${from} spectacularily broke the CEL parser; we would very much appreciate you opening a Github issue.`
```

Comment:
lol is this too cute of an error message?

We should at least include why we think it's broken - e.g. too many nested params?

Reply:
This is an escape hatch that users ~*~should never see~*~; it's there to prevent infinite loops in string interpolation which I can't see could ever happen

Reply:
oh no

Reply:
it could happen

Reply:
Okay, it can't happen now, so we can get rid of this cute message.